classes/client/s3_client.php

```diff
@@ -331,6 +341,55 @@ public function define_client_section($mform, $config) {
 
         $mform->addElement('select', 's3_region', get_string('settings:aws:region', 'tool_objectfs'), $regionoptions);
         $mform->addHelpButton('s3_region', 'settings:aws:region', 'tool_objectfs');
+
+        $this->define_proxy_config($mform, $config);
+
         return $mform;
     }
+
+    private function define_proxy_config($mform, $config) {
+
+        $mform->addElement('header', 'proxyheader', get_string('settings:proxyheader', 'tool_objectfs'));
+        $mform->setExpanded('proxyheader');
+
+        $mform->addElement('text', 'proxyserver', get_string('settings:proxyserver', 'tool_objectfs'));
+        $mform->addHelpButton('proxyserver', 'settings:proxyserver', 'tool_objectfs');
+        $mform->setType("proxyserver", PARAM_TEXT);
+
+        $mform->addElement('text', 'proxyport', get_string('settings:proxyport', 'tool_objectfs'));
+        $mform->addHelpButton('proxyport', 'settings:proxyport', 'tool_objectfs');
+        $mform->setType("proxyport", PARAM_TEXT);
+
+        $mform->addElement('text', 'proxyusername', get_string('settings:proxyusername', 'tool_objectfs'));
+        $mform->addHelpButton('proxyusername', 'settings:proxyusername', 'tool_objectfs');
+        $mform->setType("proxyusername", PARAM_TEXT);
+
+        $mform->addElement('password', 'proxypassword', get_string('settings:proxypassword', 'tool_objectfs'));
+        $mform->addHelpButton('proxypassword', 'settings:proxypassword', 'tool_objectfs');
+        $mform->setType("proxypassword", PARAM_TEXT);
+    }
+
+    private function construct_proxy_string($config) {
+        $proxy = '';
+        if (!empty($config->proxyserver)) {
+            if (!empty($config->proxyusername)) {
+                $arrayproxy = explode("//", $config->proxyserver);
```

Comment:
Would be good to add some comments here to understand what could com as $config->proxyserver and explain why we have som complicated code below. Test are appreciated as well ;)